pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -134,31 +134,36 @@ func IsTrustedCABundleUpdated(ctx context.Context, cli client.Client, dscInit *d
 }
 
 func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.Logger, dscInit *dsci.DSCInitialization, managementStateChanged bool) error {
-	switch dscInit.Spec.TrustedCABundle.ManagementState {
-	case operatorv1.Managed:
-		log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update " + CAConfigMapName)
-		istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
-		if err != nil {
-			return err
-		}
-
-		if istrustedCABundleUpdated || managementStateChanged {
-			if err := AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit); err != nil {
-				log.Error(err, "error adding configmap to all namespaces", "name", CAConfigMapName)
+	if dscInit.Spec.TrustedCABundle != nil {
+		switch dscInit.Spec.TrustedCABundle.ManagementState {
+		case operatorv1.Managed:
+			log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update " + CAConfigMapName)
+			istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
+			if err != nil {
 				return err
 			}
-		}
-	case operatorv1.Removed:
-		log.Info("Trusted CA Bundle injection is set to `Removed` state. Reconciling to delete all " + CAConfigMapName)
-		if err := RemoveCABundleConfigMapInAllNamespaces(ctx, cli).ErrorOrNil(); err != nil {
-			log.Error(err, "error deleting configmap from all namespaces", "name", CAConfigMapName)
-			return err
+
+			if istrustedCABundleUpdated || managementStateChanged {
+				if err := AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit); err != nil {
+					return fmt.Errorf("failed adding configmap %s to all namespaces: %w", CAConfigMapName, err)
+				}
+			}
+		case operatorv1.Removed:
+			log.Info("Trusted CA Bundle injection is set to `Removed` state. Reconciling to delete all " + CAConfigMapName)
+			if err := RemoveCABundleConfigMapInAllNamespaces(ctx, cli).ErrorOrNil(); err != nil {
+				return fmt.Errorf("error deleting configmap %s from all namespaces %w", CAConfigMapName, err)
+			}
+		case operatorv1.Unmanaged:
+			log.Info("Trusted CA Bundle injection is set to `Unmanaged` state. " + CAConfigMapName + " configmaps are no longer managed by operator")
 		}
 
-	case operatorv1.Unmanaged:
-		log.Info("Trusted CA Bundle injection is set to `Unmanaged` state. " + CAConfigMapName + " configmaps are no longer managed by operator")
+		return nil
 	}
 
+	log.Info("Trusted CA Bundle is not configed in DSCI, same as default to `Removed` state. Reconciling to delete all " + CAConfigMapName)
+	if err := RemoveCABundleConfigMapInAllNamespaces(ctx, cli).ErrorOrNil(); err != nil {
```

Comment:
Not part of this PR, but shouldn't `RemoveCABundleConfigMapInAllNamespaces` return an error instead?

Reply:
updated